### ISSUE TITLE
Error parsing multi-line quoted text containing asterisk.

### ISSUE THREAD

**Issue by tubedude** [pragdave/earmark#48]
This error came up in a PR in [ericmj/decimal](https://github.com/ericmj/decimal/pull/23) (The comment is outdated and therefore hidden).

[This code](https://github.com/ericmj/decimal/blob/master/lib/decimal.ex#L5-L6):

```elixir
defmodule Decimal do
  @moduledoc """
  Decimal arithmetic on arbitrary precision floating-point numbers.

  A number is represented by a signed coefficient and exponent such that: `sign
  * coefficient * 10^exponent`. All numbers are represented and calculated
  exactly, but the result of an operation may be rounded depending on the
  context the operation is performed with, see: `Decimal.Context`. Trailing
  zeros in the coefficient are never truncated to preserve the number of
  significant digits unless explicitly done so.
```

generates the following html:

```html
<section id="moduledoc" class="docstring">
<p>Decimal arithmetic on arbitrary precision floating-point numbers.</p>
<p>A number is represented by a signed coefficient and exponent such that: `sign</p>
<ul>
<li>coefficient * 10^exponent`. All numbers are represented and calculated
exactly, but the result of an operation may be rounded depending on the
context the operation is performed with, see: <a href="Decimal.Context.html"><code class="inline">Decimal.Context</code></a>. Trailing
zeros in the coefficient are never truncated to preserve the number of
significant digits unless explicitly done so.
</li>
</ul>
```

![decimal_ _decimal_v1_1_1-dev-error](https://cloud.githubusercontent.com/assets/1635127/11283400/afbf42e2-8eeb-11e5-8372-f41924b4cac4.png)

While the correct should be something like this:

```html
<section id="moduledoc" class="docstring">
<p>Decimal arithmetic on arbitrary precision floating-point numbers.</p>
<p>A number is represented by a signed coefficient and exponent such that: <code class="inline">sign
coefficient * 10^exponent</code>. All numbers are represented and calculated
exactly, but the result of an operation may be rounded depending on the
context the operation is performed with, see: <a href="Decimal.Context.html"><code class="inline">Decimal.Context</code></a>. Trailing
zeros in the coefficient are never truncated to preserve the number of
significant digits unless explicitly done so.
</p>
```
![decimal_ _decimal_v1_1_1-dev](https://cloud.githubusercontent.com/assets/1635127/11283443/ecce4e4e-8eeb-11e5-8ba4-5ef36bed38c9.png)

**Comment by RobertDober** [comment]
This is not an error but a design choice as explained in the README. 

Two workarounds come into mind immediately

```iex
iex(7)> text = ~s/Hello `sign\n \\* one`/
"Hello `sign\n \\* one`"
iex(8)> Earmark.to_html text             
"<p>Hello `sign\n * one`</p>\n"
iex(9)> text = ~s/Hello `sign\n &nbsp;* one`/
"Hello `sign\n &nbsp;* one`"
iex(10)> Earmark.to_html text                 
"<p>Hello `sign\n &nbsp;* one`</p>\n"
iex(11)> text = ~s/Hello `sign\n &nbsp;* one`/
```

For what it is worth, about 30% of the markdown implementations on Babelmark (http://johnmacfarlane.net/babelmark2/?text=%60sign%0A*+expression%60) agree with Dave's interpretation.
Not sure how the second renders. Maybe it is just best to rearrange the doc so that the line does not start with an `*` which is more readable anyway.

**Comment by RobertDober** [comment]
oops that reveals another issue

**Comment by RobertDober** [comment]
@tubedude sorry if I missread your issue, you meant #49 not the `<ul> issue` right?

**Comment by tubedude** [comment]
Yes @RobertDober I think this issue is related to the backquotes enclosing a multiline text.
I agree with you that this issue and #49 are the same.

**Comment by RobertDober** [comment]
@pragdave I just pushed a development branch that fixes #48, #49 & #51. However the code is not one that I would happily merge it into master.
I need your wisdom here :). But is this the right place to discuss the code quality and design choices I have made? (or rather not made, I tried to respect your design as much as possible)

**Comment by RobertDober** [comment]
Closed with 55828d1131581554712849bbacd82a88568ba66c
